src/main/java/jenkins/bouncycastle/api/PEMEncodable.java

```diff
@@ -179,15 +179,15 @@ public static PEMEncodable decode(@Nonnull String pem, @Nullable final char[] pa
             // handle supported PEM formats.
             if (object instanceof PEMEncryptedKeyPair) {
                 if (passphrase != null) {
-                    PEMDecryptorProvider dp = new JcePEMDecryptorProviderBuilder().build(passphrase);
+                    PEMDecryptorProvider dp = new JcePEMDecryptorProviderBuilder().setProvider(BOUNCY_CASTLE_PROVIDER).build(passphrase);
                     PEMEncryptedKeyPair ekp = (PEMEncryptedKeyPair) object;
-                    return new PEMEncodable(kConv.getKeyPair(ekp.decryptKeyPair(dp)));
+                    return new PEMEncodable(kConv.getPrivateKey(ekp.decryptKeyPair(dp).getPrivateKeyInfo()));
```

Comment:
this would mean we would be unable to recover a public key given a keypair would it not?

Reply:
I will revert this change here, just testing. It should be as it is nowadays.